File: frontend/types/job.ts
export interface Job {
    JobID: number;
    Title: string;
    Department: string;
    Location: string;
    EmploymentType: string;
    Salary: string;
    Description: string;
    Requirements: string;
    Status: string;
    CreatedAt: string;
  }
  
  export interface JobCreate {
    Title: string;
    Department: string;
    Location: string;
    EmploymentType: string;
    Salary: string;
    Description: string;
    Requirements: string;
    Status: string;
  }
  
  export interface JobUpdate extends JobCreate {}

File: backend/app/schemas/job.py
from pydantic import BaseModel
from datetime import datetime


class JobBase(BaseModel):
    Title: str
    Department: str
    Location: str
    EmploymentType: str
    Salary: str
    Description: str
    Requirements: str
    Status: str = "Open"


class JobCreate(JobBase):
    pass


class JobUpdate(JobBase):
    pass


class JobResponse(JobBase):
    JobID: int
    CreatedAt: datetime

    class Config:
        from_attributes = True